### PR TITLE
Add data object to Messages class

### PR DESCRIPTION
**Pull request by AFRUITPIE**
**Please describe the changes this PR makes and why it should be merged:**
Although adding `data` as another field of `Message` uses more memory, it is valuable for more easily extending `Message` in TypeScript. 

For example, extending `Message` in TypeScript can be done with: 

```
class DogMessage extends Message {
  woof(): void {
    this.channel.send("Woof!")
  }
}

const client = new Client();

client.on("message", (message: Message) => {
  let dogMessage = new DogMessage(message.channel, message.data, message.client);
  dogMessage.woof()
});
```

This would require adding:
```
export class Message {
  ...
  public data: object;
  ...
}
```
to the typings file.

**Semantic versioning classification:**  
- [ ] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by appellation** [comment]
The master branch already has built-in structure extenstion and typescript supports type extensions already. 

I'm confused about how adding a raw data property helps with this in any way, and why it should only be applied to message structures if there's some reasoning I've overlooked.

**Comment by AFRUITPIE** [comment]
Maybe I'm overlooking something, but extending `Message` in TypeScript is difficult to do because of `Message`'s constructor. The `data` parameter seems next to impossible to recreate when getting a message from the `client.on("message", message => { ... })` callback.

Doing `client.on("message", (message: DogMessage) => { ... }` does nothing because TypeScript has type assertion rather than type casting.

**Comment by appellation** [comment]
Let me rephrase.

Structure extension is trivial with the new master branch structures API, and typescript already supports module augmentation. 

It sounds like you're attempting to extend structures by creating new ones, which is very memory inefficient. The nature of JavaScript and by extension Typescript allows for easy extension and modification of modules, which should be used in preference to creating new modules. Consider using either prototype modification or the new structures API in conjunction with Typescript module augmentation. Adding a property to support poor coding practice is a -1 from me.
